tracer/src/Datadog.Trace/Vendors/IndieSystem.Text.RegularExpressions/System/Text/SegmentStringBuilder.cs

```diff
@@ -60,9 +60,14 @@ private void GrowAndAdd(ReadOnlyMemory<char> segment)
             const int DefaultArraySize = 256;
             int newSize = array.Length == 0 ? DefaultArraySize : array.Length * 2;
 
+#if NO_ARRAY_POOL
+            ReadOnlyMemory<char>[] newArray = _array = new ReadOnlyMemory<char>[newSize];
```

Comment:
I genuinely can't follow this 😬 I get that it's "mirroring" the one below (which I also despise), but I thought that `ReadOnlyMemory<char>` _is_ an abstraction over `char[]` (etc), so isn't `ReadOnlyMemory<char>` a `char[][]`? 🤔 🤷 

Also, `x = y = z`? 🤮  (again, .NET src's fault not yours, but would rather we didn't use it in code that we've written 😉)

Reply:
I have removed the double assignment upstream and re-imported, it isn't needed here.

It's char[][] or easier to think of as List<String>, that get's build up to one string in the end.